### ISSUE TITLE
CSS Minifier breaks page style

### ISSUE THREAD

**Issue by cebe** [yiisoft-contrib/yiiframework.com#234]
```
gulp build
```
![bildschirmfoto von 2017-10-16 15-37-45](https://user-images.githubusercontent.com/189796/31614901-1b82e36e-b288-11e7-8254-002a69506647.png)

```
gulp build --production
```

![bildschirmfoto von 2017-10-16 15-41-31](https://user-images.githubusercontent.com/189796/31615030-823c671a-b288-11e7-8acf-9ebe258a5051.png)

**Comment by samdark** [comment]
Any errors?

**Comment by cebe** [comment]
```
# gulp build
[09:21:20] Using gulpfile /var/www/yiiframework.demo/gulpfile.js
[09:21:20] Starting 'build'...
[09:21:20] Starting 'clean'...
[09:21:20] Finished 'clean' after 14 ms
[09:21:20] Starting 'sprites'...
[09:21:24] Finished 'sprites' after 3.26 s
[09:21:24] Starting 'styles'...
[09:21:24] Starting 'scripts'...
[09:21:24] Starting 'fonts'...
[09:21:28] Finished 'fonts' after 3.8 s
[09:21:28] gulp-notify: [Gulp notification] Scripts task complete
[09:21:28] gulp-notify: [Error in notifier] Error in plugin 'gulp-notify'
Message:
    not found: notify-send
[09:21:28] gulp-notify: [Gulp notification] Scripts task complete
[09:21:28] gulp-notify: [Error in notifier] Error in plugin 'gulp-notify'
Message:
    notify-send must be installed on the system.
[09:21:28] Finished 'scripts' after 4.52 s
[09:21:30] gulp-notify: [Gulp notification] Styles task complete
[09:21:30] gulp-notify: [Error in notifier] Error in plugin 'gulp-notify'
Message:
    notify-send must be installed on the system.
[09:21:30] gulp-notify: [Gulp notification] Styles task complete
[09:21:30] gulp-notify: [Error in notifier] Error in plugin 'gulp-notify'
Message:
    notify-send must be installed on the system.
[09:21:30] Finished 'styles' after 6.29 s
[09:21:30] Finished 'build' after 9.94 s
```

```
# gulp build --production
[09:21:56] Using gulpfile /var/www/yiiframework.demo/gulpfile.js
[09:21:56] Starting 'build'...
[09:21:56] Starting 'clean'...
[09:21:56] Finished 'clean' after 22 ms
[09:21:56] Starting 'sprites'...
[09:21:59] Finished 'sprites' after 3.61 s
[09:22:00] Starting 'styles'...
[09:22:00] Starting 'scripts'...
[09:22:00] Starting 'fonts'...
[09:22:15] gulp-notify: [Gulp notification] Scripts task complete
[09:22:15] gulp-notify: [Error in notifier] Error in plugin 'gulp-notify'
Message:
    not found: notify-send
[09:22:15] Finished 'scripts' after 16 s
[09:22:29] Finished 'fonts' after 29 s
[09:22:29] gulp-notify: [Gulp notification] Styles task complete
[09:22:29] gulp-notify: [Error in notifier] Error in plugin 'gulp-notify'
Message:
    notify-send must be installed on the system.
[09:22:29] Finished 'styles' after 29 s
[09:22:29] Finished 'build' after 33 s
```

**Comment by cebe** [comment]
I suspect a bug in the css minifier. Need to investigate that.

**Comment by jacmoe** [comment]
Could perhaps try and update the css uglifier?

**Comment by cebe** [comment]
`npm update gulp-cssnano cssnano` fixed it, thanks! 

however... I now have `cssnano@3.10.0` locally and `cssnano@3.5.0` on the server. Is there a way to enforce a version like with composer.lock?

**Comment by samdark** [comment]
https://docs.npmjs.com/files/package-lock.json

**Comment by cebe** [comment]
> package-lock.json is automatically generated for any operations where npm modifies either the node_modules tree, or package.json

it does not generate one for me...

**Comment by samdark** [comment]
npm should be latest.

**Comment by cebe** [comment]
https://wpartisan.me/tutorials/create-npm-lock-file-packages-shrinkwrap

> tldr: You can lock down your exact npm package versions by using ‘npm’shrinkwrap’

**Comment by cebe** [comment]
solved on the `search` branch.

**Comment by rob006** [comment]
> however... I now have cssnano@3.10.0 locally and cssnano@3.5.0 on the server. Is there a way to enforce a version like with composer.lock?

https://yarnpkg.com/lang/en/

**Comment by jacmoe** [comment]
Yarn is cool, but when npm 5 hits the main stream, it will stop being
attractive :)

2017-10-18 18:49 GMT+02:00 Robert Korulczyk <notifications@github.com>:

**Comment by cebe** [comment]
I wanted to solve this as simple as possible, without switching the existing stack ;)

**Comment by rob006** [comment]
@cebe You only need to replace `npm install` call with `yarn install` - you get native lock support, faster installs and more convenient syntax for node dependency management. 

@jacmoe I hardly believe it. It is still faster, more convenient and more reliable (due to npm fragmentation) than npm 5. It already has twice more stars on GitHub than npm. :D

**Comment by cebe** [comment]
> @cebe You only need to replace npm install call with yarn install - you get native lock support, faster installs and more convenient syntax for node dependency management.

I'll probably try it when I find some time for it but I need to install it locally and on the server. Consindering bad experience with nodejs in the past I preferred to not waste my time on that.

**Comment by jacmoe** [comment]
@rob006 I already have npm 5.x installed, and it gives me lock support, and that is all I need! 😄 

Even though my policy is to avoid installing anything JavaScript related just because it's new, I was "tricked" into installing Yarn :p
I am just tired of having to learn new shit every month, that's all :D

**Comment by allyraza** [comment]
should not we bump the npm to v5 instead of using npm-shrinkwrap so its fixed for good instead of updating it again.

**Comment by cebe** [comment]
what wrong with npm-shrinkwrap?

**Comment by samdark** [comment]
Solved.
